bin/validate_sample_meta.pl

```diff
@@ -0,0 +1,183 @@
+#! /usr/bin/perl
+
+# perl-5.16.3 -I /software/CGP/canpipe/live/lib/perl5 validate_smaple_meta.pl
+use strict;
+use warnings FATAL => 'all';
+use autodie qw(:all);
+use Getopt::Long;
+use Pod::Usage qw(pod2usage);
+use Const::Fast qw(const);
+use Carp;
+
+use Spreadsheet::Read qw(ReadData);
+use Spreadsheet::WriteExcel; # to write xls output
+use Excel::Writer::XLSX; # to write xlsx output
+
+use Sanger::CGP::NgsQc::ValidateSampleMetaAndBamHeader;
+
+my $opts = setup_options();
+my $input = $opts->{'i'};
+my $output = $opts->{'o'};
+my $check_all = 'n';
+if (defined $opts->{'a'}) {
+  $check_all = 'y';
+}
+#print "check_all setting: $check_all.\n";
+my $format;
+
+if (defined $opts->{'f'} && $opts->{'f'} =~ m/^xls$|^xlsx$|^tsv$/i) {
+  $format = $opts->{'f'};
+} elsif (! defined $opts->{'f'}) {
+  $format = 'tsv';
+} else {
+  die("--------\nError: Does not recognise the specified file format, accept values are: 'tsv', 'xls' and 'xlsx'.\n--------\n");
+}
+
+my @in;
+### read in input as array of lines (concatenated columns with tabs)
+if ($format eq 'xlsx'|| $format eq 'xls') {
+  my $book = ReadData ($input);
+  my @rows = Spreadsheet::Read::rows($book->[1]);
+  foreach my $row (@rows) {
+    foreach my $cell (@$row) {
+      $cell = '' if (! defined $cell);
+      # print $cell."\n";
+    }
+    push @in, (join "\t", @$row);
+  }
+} else { # assume it's a tsv
+  open (my $FH, '<', $input) or croak("--------\nError trying to open input file $input: $!\n--------\n");
+  chomp(@in = <$FH>);
+  close $FH;
+}
+
+my ($valid_results, $valid_status) = Sanger::CGP::NgsQc::ValidateSampleMetaAndBamHeader::validate_samples(\@in, $check_all);
+# print "below is the out:\n";
+# foreach my $line (@$valid_results) {print "$line\n";}
+
+### output into the same format again on failed validation, an extra tsv if (input is excel) on a successful validation!
+if ($format eq 'xlsx') {
+  # rename output, if it does not ends with .xlsx
+  if ($output !~ /\.xlsx$/) {
+    $output .= '.xlsx';
+  }
+  my $workbook = Excel::Writer::XLSX->new($output);
+  my $worksheet = $workbook->add_worksheet();
+  my $row = 0;
+  foreach my $line (@$valid_results) {
+    my @array = split("\t", $line);
+    $worksheet->write_row($row, 0, \@array);
+    $row++;
+  }
+  if ($valid_status eq "green") {
```

Comment:
This block is repeated many times, it should be a function (within the object) which dynamically can change the file type.

Ideally `$valid_status` should be `[0|1]`, or a names number constant if more than 2 states needed.

Is not interpolation is required in a string use `'` not `"`